src/elements/content-explorer/ContentExplorer.js

```diff
@@ -525,16 +529,23 @@ class ContentExplorer extends Component<Props, State> {
      * @param {Object} collection item collection object
      * @return {void}
      */
-    searchSuccessCallback = (collection: Collection) => {
+    searchSuccessCallback = async (collection: Collection) => {
+        const { features }: Props = this.props;
         const { selected }: State = this.state;
+        const { items } = collection;
 
         // Unselect any rows that were selected
         this.unselect();
 
         // Close any open modals
         this.closeModals();
 
-        this.updateCollection(collection, selected);
+        if (isFeatureEnabled(features, 'contentExplorer.gridView.enabled')) {
+            const itemsWithThumbnails = await this.populateItemsWithThumbnails(items);
```

Comment:
Can the logic inside `populateItemsWithThumbnails` live in `this.updateCollection` so it doesn't need to be called from each success callback?

Reply:
Still looking into this.